### PR TITLE
DOC: fix method signatures in "array subclasses"

### PR DESCRIPTION
**Pull request by mgeier**
- Change "`.. function::`" -> "`.. method::`"
- Remove "self" argument
- Change "self" to "obj" in `__array_finalize__`

### CODE REVIEW

File: doc/source/reference/arrays.classes.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ Special attributes and methods
 
 Numpy provides several hooks that classes can customize:
 
-.. function:: class.__numpy_ufunc__(self, ufunc, method, i, inputs, **kwargs)
+.. method:: class.__numpy_ufunc__(ufunc, method, i, inputs, **kwargs)
 
    .. versionadded:: 1.10
```
Comment (mhvk) [inline doc/source/reference/arrays.classes.rst]: Happened to notice this: actually not added in 1.10 in the end -- should become 1.11.

Comment (mgeier) [inline doc/source/reference/arrays.classes.rst]: OK, I wasn't aware of that.

Am I supposed to change this in my PR?

**Comment by charris** [comment]
@mgeier Yes, just make the change, use `git rebase -i HEAD^^` to squash the two commits into one, then force push `git push -f origin <branch-name>`.

**Comment by mgeier** [comment]
OK, done.

**Comment by charris** [comment]
Thanks @mgeier .

**Pull request merged**